orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -921,35 +921,32 @@ IMPLEMENT_GRADIENT_BUILDER(GetFastGeluGradient) {
 }
 
 IMPLEMENT_GRADIENT_BUILDER(GetSendGradient) {
-  NodeAttributes new_attrs;
-  const auto& old_attrs = SrcNodeAttributes();
- 
-  new_attrs["src"] = MakeAttribute("src", old_attrs.at("dst").i());
-  new_attrs["dst"] = MakeAttribute("dst", old_attrs.at("src").i());
-  new_attrs["tag"] = MakeAttribute("tag", old_attrs.at("tag").i());  // keep tag the same for now
-  new_attrs["element_type"] = MakeAttribute("element_type", old_attrs.at("element_type").i());
+  std::vector<ArgDef> out_args;
+  for (int i = 1; i < GetSrcNodeInputSize(); ++i) {
+    out_args.push_back(GI(i));  // i=1 for Signal, the rest Data
+  }
 
   return std::vector<NodeDef>{
       NodeDef("Recv",
-              {O(0)},
-              {GI(0), GI(1)},
-              new_attrs)};
+              {I(0),O(0)},  // {Remote, Signal} 
+              out_args,
+              SrcNodeAttributes())};
```

Comment:
>SrcNodeAttributes [](start = 14, length = 17)

need to copy element types from Send node?

Reply:
Recv needs element_types for type inference.
Send needs element_types for its gradient, which is Recv.
Though Recv's gradient is Send, and doesn't care element_types, I will still clone it for consistency.